pkg/cmd/roachtest/main_test.go

```diff
@@ -167,6 +172,188 @@ func Test_updateSpecForSelectiveTests(t *testing.T) {
+			// So, the Skip is either blank or has a value as "<test name> skipped"
+			if _, ok := toBeSelectedTestsMap[s.Name]; ok {
+				allTestsToBeSelected++
+				assert.True(t, s.Skip == "" || s.Skip == getSkippedMessage(s.Name), s.Skip)
+			}
+		}
+		require.Equal(t, countOfSpecSkippedTests, skippedTestsInSpec)
+		require.Equal(t, allTestsToBeSelected, len(toBeSelectedTestsMap))
+	})
+}
+
+func getSkippedMessage(testName string) string {
+	return fmt.Sprintf("%s skipped", testName)
+}
+
+// getRandomisedTests returns "totalTests" number of randomised tests.
+// Out of the totalTests,
+// > 0-90% of the tests are added as tests in specs
+// > 10-100% of the tests are added as snowflake returned tests
+// So, the overlap is 80%
+// The above tests are shuffled and randomly added different conditions.
+// The function returns:
+// > List of test specs
+// > The rows to be returned by the snowflake as [][]string
+// > The rows as *sqlmock.Rows
+// > The map of tests that will be selected based on the percent criteria
+func getRandomisedTests(
+	t *testing.T, totalTests int,
+) ([]registry.TestSpec, [][]string, *sqlmock.Rows, map[string]struct{}) {
+	testNames, err := generateUniqueStrings(totalTests, 10, 100)
+	require.Nil(t, err)
+	// As mentioned,
+	// 0% to 90% of tests are added to test spec
+	// 10% to 100% of tests are added to snowflake tests
+	// So, the overlap is 80%
+	sfStart := int(float32(len(testNames)) * 0.1)
+	specEnd := int(float32(len(testNames)) * 0.9)
+	// testForSF are the name of tests which are returned by SF query
+	testForSF := shuffleStrings(t, testNames[sfStart:])
+	// testForSpecs are the name of tests which are in the test spec
+	testForSpecs := shuffleStrings(t, testNames[:specEnd])
+	// commonSuccessTestMap contains all test that are present in both specs and snowflake minus the tests that are
+	// marked as "selected=true"
+	commonSuccessTestMap := make(map[string]struct{})
+	for _, tn := range testNames[sfStart:specEnd] {
+		// first add all the common tests. The tests marked as "selected=true" will be removed later
+		commonSuccessTestMap[tn] = struct{}{}
+	}
+	rows := sqlmock.NewRows(testselector.AllRows)
+	// data is created and returned for easier assertions
+	data := make([][]string, len(testForSF))
+	// testSelected is a randomised value which is the number of tests that are marked as "selected=true"
+	testSelected, err := rand.Int(rand.Reader, big.NewInt(int64(0.6*float32(len(testForSF)))))
+	require.Nil(t, err)
+	for i, sfn := range testForSF {
+		selected := "no"
+		lastFailureIsPreempt := "no"
+		if i < int(testSelected.Int64()) {
+			// mark the tests as selected sequentially
+			selected = "yes"
+			// delete the entry from the success map as this is selected.
+			delete(commonSuccessTestMap, sfn)
+			// from the selected tests mark a few tests as "last_failure_is_preempt=yes"
+			testPreempted, err := rand.Int(rand.Reader, big.NewInt(testSelected.Int64()))
```

Comment:
nit: maybe `numTestsPreempted` so it's clearer it's not the set of tests preempted and just the count?